giraffe/README.md

```diff
@@ -0,0 +1,341 @@
+
+  <pre>
+  const config = {
+    // ...
+  
+    valueFormatters: {
+      _time: (t) => new Date(t).toLocaleTimeString(),  // ECMAScript time to human-readable time stamp
+      _value: (num) => num.toFixed(2),                 // values fixed to 2 decimal places
+    },
+  
+    // ...
+  }
+  </pre>
+
+- **xAxisLabel**: _string. Optional._ Uses **tickFont**, **tickFontColor**. Name to display below the x-axis.
+
+- **yAxisLabel**: _string. Optional._ Uses **tickFont**, **tickFontColor**. Name to display to the left of the y-axis.
+
+### Data properties
+
+- **table**: _Object. **Required**._ A data object produced by [Giraffe's utility functions](#utility-functions). Houses the data and getters and setters for that data, for the `<Plot>`.
+
+  - the `table` property of the return value from the `fromFlux` utility function.
+  - the return value from the `fromRows` utility function.
+  - the return value from the `newTable` utility function.
+
+- **layers**: _array[Object, ...]. **Required**._ An array of [LayerConfig objects](#layerconfig). These objects are customizations specific to a type of `<Plot>`. Currently, Giraffe supports only one pre-defined `<Plot>` type with any number of "custom layers". Custom layers are not pre-defined in Giraffe, but created through a callback render function in the configuration.
+
+- **xScale**: _"linear" | "log". Optional._ Sets the scaling of the x-axis by selecting the correct scaling function to be used by Layers. Linear scaling means the same distance between ticks represent the same increase in value. Logarithmic (log) scaling means the same distance between ticks can represent an exponential increase in value, used as a way to display data with a very wide range of values in a compact space.
+
+- **yScale**: _"linear" | "log". Optional._ Sets the scaling of the y-axis by selecting the correct scaling function to be used by Layers. Linear scaling means the same distance between ticks represent the same increase in value. Logarithmic (log) scaling means the same distance between ticks can represent an exponential increase in value, used as a way to display data with a very wide range of values in a compact space.
+
+- **xDomain**: _array[min, max]. Optional._ The x domain of the plot can be explicitly set with numbers denoting a _min_ and _max_. If this option is passed, then `<Plot>` is operating in a "controlled" mode, where it always uses the passed x domain. Any brush interaction with the `<Plot>` that should change the x domain will call the `onSetXDomain` option when the component is in controlled mode. Double clicking the plot will call `onResetXDomain`. If the `xDomain` option is not passed, then the component is "uncontrolled". It will compute, set, and reset the `xDomain` automatically.
```

Comment:
```suggestion
- **xDomain**: _array[min, max]. Optional._ The x domain of the plot can be explicitly set with numbers denoting a _min_ and _max_. If this option is passed, then `<Plot>` is operating in a "controlled" mode, where it always uses the passed x domain to set the minimum and maximum value of the x axis. Any brush interaction with the `<Plot>` that should change the x domain will call the `onSetXDomain` option when the component is in controlled mode. Double clicking the plot will call `onResetXDomain`. If the `xDomain` option is not passed, then the component is "uncontrolled". It will compute, set, and reset the `xDomain` automatically.
```

Reply:
Also, should we say something about the fact that if a user passes an `xDomain`, it is required that the value have both a `min` and `max` value? So that way users don't input: `[null, 100]`, `[0, null]`, `[0]`, etc....

Reply:
Good call 🎉